.lintstagedrc.js

```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  "frontend/*.{scss,css}": "stylelint --fix",
+  "frontend/*.{ts,tsx,js,jsx,scss,css}": "prettier --write",
+  "frontend/*.{ts,tsx,js,jsx}": (filenames) =>
+    `next lint --fix --file ${filenames
+      .map((file) => file.split(process.cwd())[1])
+      .join(" --file ")}`,
```

Comment:
Sorry for taking a while to get back to this - had a pile of other work too.

> I'm running prettier --write on all .md files, which will include the docs directory. That may be a mistake. Otherwise, I've restricted JS linters to the frontend folder. You may need to check that this did not break anything.

Hmm, I tried a bunch of things, but I can't seem to get it to work. The first cause was that these patterns didn't match staged files - I had to change them from `frontend/*.` to `frontend/**/*.`.

After that, lint-stage at least gets trigged if I modify a frontend file (e.g. replacing `"` by `'` in an import in `frontent/src/components/Banner.tsx`). Then Prettier appears to run correctly, but it hangs on linting:

https://user-images.githubusercontent.com/4251/186926483-5e7c99ab-432e-4c4c-91cc-8ec290dbece5.mp4

At first I thought that might be related to it not running in `/frontend`, but then Prettier running successfully makes me reject that thought. So not sure what's going on :/ Might have to revert back to just running in Husky's pre-commit again.

Running Prettier on all Markdown files seems fine by me.

Reply:
Thanks for the example that will trigger `prettier`. I couldn't find any simple edits that `next lint` would fix, however.

I updated the pattern to `frontend/**/*`. It took me a while to figure out how to even run `next lint` manually. When I ran `npx next lint` in the root project folder, not `frontend`, it prompted:

```
Need to install the following packages:
  next
Ok to proceed? (y) 
```

I suspect that is why the command was hanging - it was waiting for someone to type `y`, but the output was swallowed by `lint-staged`.

I updated `next lint --fix ...` to `next lint frontend --fix ...` to set the base directory, and that seemed to work.

Reply:
... and I rebased on `main`, which had a handful of `black` and `mypy` issues

Reply:
Huh. I have no idea why `next lint frontend` would work (I can't find any documentation of `next` that specifies the ability to pass `frontend`, and I also don't know why it wouldn't ask to install `next` when run with that parameter), but it does seem to work.

(I tested `next lint` fixes by replacing `===` with `==` in the comparison with `typeof` on line 315 of `/frontend/src/apiMocks/handlers.ts`.)

Reply:
> Huh. I have no idea why next lint frontend would work (I can't find any documentation of next that specifies the ability to pass frontend, and I also don't know why it wouldn't ask to install next when run with that parameter), but it does seem to work.

I can't find the docs online. I used:

```
npx next lint --help

      Description
        Run ESLint on every file in specified directories. 
        If not configured, ESLint will be set up for the first time.

      Usage
        $ next lint <baseDir> [options]      

      <baseDir> represents the directory of the Next.js application.
      If no directory is provided, the current directory will be used.
      
      ...
```

Reply:
Oh neat, didn't think of that!